File: go/605_Can_Place_Flowers.go
func canPlaceFlowers(flowerbed []int, n int) bool {
    for i:= 0; i < len(flowerbed); i++ {
        var isCurrentPlaceClear = flowerbed[i] == 0
        var isPreviousPlaceClear = i == 0 || flowerbed[i - 1] == 0
        var isNextPlaceClear = i == len(flowerbed) - 1 || flowerbed[i + 1] == 0
        if isCurrentPlaceClear && isPreviousPlaceClear && isNextPlaceClear {
            flowerbed[i] = 1;
            n--;
            i++;
        }
    }

    return n <= 0;
}

File: python/605_Can_Place_Flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        for i, num in enumerate(flowerbed):
            isCurrentPlaceClear = flowerbed[i] == 0
            isPreviousPlaceClear = i == 0 or flowerbed[i - 1] == 0
            isNextPlaceClear = i == len(flowerbed) - 1 or flowerbed[i + 1] == 0
            if isCurrentPlaceClear and isPreviousPlaceClear and isNextPlaceClear:
                flowerbed[i] = 1
                n -= 1
        return n <= 0
